app/src/ereferrals/java/org/eyeseetea/malariacare/data/database/datasources/SettingsDataSource.java

```diff
@@ -1,22 +1,32 @@
 package org.eyeseetea.malariacare.data.database.datasources;
 
 import android.annotation.TargetApi;
+import android.content.Context;
+import android.content.SharedPreferences;
 import android.content.res.Resources;
 import android.os.Build;
+import android.preference.PreferenceManager;
 
+import org.eyeseetea.malariacare.R;
 import org.eyeseetea.malariacare.data.database.utils.PreferencesState;
 import org.eyeseetea.malariacare.domain.boundary.repositories.ISettingsRepository;
+import org.eyeseetea.malariacare.domain.entity.MediaListMode;
 import org.eyeseetea.malariacare.domain.entity.Settings;
 
 import java.util.Locale;
 
 public class SettingsDataSource implements ISettingsRepository {
+    private final Context context;
+
+    public SettingsDataSource() {
+        context = PreferencesState.getInstance().getContext();
```

Comment:
as we have already spoken in several meetings with @ifoche and @manuelplazaspalacio, in new code should avoid using static context saved in preference state because is problematic for memory leaks, if you need context, refactor to receive it in the constructor